File: 1250-LongestCommonSubsequence/1250-LongestCommonSubsequence.py
# Last updated: 2/23/2026, 6:44:44 PM
class Solution:
    def longestCommonSubsequence(self, text1: str, text2: str) -> int:
        len1 = len(text1)
        len2 = len(text2)
        if len1<=len2:
            text1, text2 = text2, text1
            len1, len2 = len2, len1

        prev_row = [0]*(len2+1)
        curr_row = [0]*(len2+1)

        for i in range(1, len1+1):
            curr_row = [0] * (len2 + 1)
            for j in range(1, len2+1):
                if text1[i-1]==text2[j-1]:
                    curr_row[j] = prev_row[j-1]+1
                else:
                    curr_row[j] = max(prev_row[j], curr_row[j-1])
            prev_row = curr_row
        
        return curr_row[len2]

File: 1250-LongestCommonSubsequence/1250-LongestCommonSubsequence.java
// Last updated: 2/23/2026, 6:44:46 PM
class Solution {
    public int longestCommonSubsequence(String text1, String text2) {
        if(text1==null || text2==null || text1.length()==0 || text2.length()==0)
            return 0;
        
        int text1_size = text1.length();
        int text2_size = text2.length();
        
        int[][] tmp = new int[text1_size+1][text2_size+1];
        for (int i=0; i<text1_size+1; i++) {
              tmp[i][0] = 0;
        }
        for(int j=0; j<text2_size+1; j++) {
            tmp[0][j] =0;
        }
        
        if(text1.charAt(0)==text2.charAt(0))
            tmp[1][1] = 1;
        else
            tmp[1][1] = 0;
        
        for(int i=1; i<text1_size+1; i++) {
            for(int j=1; j<text2_size+1; j++) {
                if(text1.charAt(i-1) == text2.charAt(j-1))
                    tmp[i][j] = tmp[i-1][j-1]+1;
                else
                    tmp[i][j] = Math.max(tmp[i-1][j], tmp[i][j-1]);
            }            
        }
        
        return tmp[text1_size][text2_size];
    }
}